ambry-network/src/main/java/com.github.ambry.network/NettyServerRequestResponseChannel.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright 2019 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.network;
+
+import com.github.ambry.rest.RestResponseChannel;
+import com.github.ambry.rest.RestUtils;
+import java.io.IOException;
+import java.util.concurrent.ArrayBlockingQueue;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+
+
+/**
+ * RequestResponseChannel for Netty based server
+ */
+public class NettyServerRequestResponseChannel implements RequestResponseChannel {
```

Comment:
there are two different ways of dealing with requests in ambry-frontend and ambry-server.
In ambry-server, we have a RequestResponseChannel, which will return requests and queue up responses. RequestHandler would poll the requests from RequestResponseChannel.

In ambry-frontend, we have a AsyncRequestResponseHandler that handles such logic.

Do we want to keep it consistent with the frontend since they both going to use Netty HTTP?

I mean let NettyServerRequestResponseChannel implement RequestResponseHandler?

Reply:
They are some logic differences. 
`AsyncRequestResponseHandler ` just triggers the `response`, the `response` is sent out by router piece by piece. 
In server, we can get a complete `response` and then send data out. 

For scalability and cleanness on server side, I prefer to use `RequestHandler`.

Reply:
It's also for compatibility with current stack. 
Because we want to support dual port in the first phase, we can't remove Refactor AmbryRequest too much.  NettyServerRequestResponseChannel is another layer for netty http server, but we can optimize it later.